File: AspectsTests/TestObject.m
//
//  TestObject.m
//  AspectsTests
//
//  Created by Yanni Wang on 13/5/20.
//  Copyright © 2020 Yanni. All rights reserved.
//

#import "TestObject.h"

@implementation TestObject

- (void)noArgsNoReturnFunc
{
    
}

- (void)execute:(void(^)(void))block
{
    block();
}

- (void)subclassOverridedFunc
{
    [super subclassOverridedFunc];
}

+ (void)classMethod
{
    
}

@end

File: iOSHookTests/TestObjects/TestObject.swift
//
//  TestObject.swift
//  iOSHookTests
//
//  Created by Yanni Wang on 14/10/19.
//  Copyright © 2019 Yanni. All rights reserved.
//

import UIKit

class TestObject: SuperTestObject {
    @objc override func simpleMethod() {
        super.simpleMethod()
    }
    @objc func executedBlock(_ block: () -> Void) {
    }
    @objc func returnParameter(_ value: Any) -> Any {
        return value
    }
    @objc class func classSimpleMethod() {
    }
}
